simple_certmanager/admin.py

```diff
@@ -82,11 +56,51 @@ class SigningRequestAdmin(admin.ModelAdmin):
     readonly_fields = ("csr",)
     actions = [download_csr]
 
+    def get_fieldsets(self, request, obj=None):
```

Comment:
I think you can do revert this and instead in the model form check if an instance is provided or not:

```py
def __init__(self, *args, **kwargs):
    super().__init__(*args,**kwargs)
    if not self.instance.pk:  # adding a new object instead of updating an existing one
        del self.fields["should_renew_csr"]
```

edit: I see that's what you're doing already. I think you can safely include the field in the static `fieldsets` and django is smart enough to not crash.

Reply:
If I hardcode `should_renew_csr` in the fieldset than it gives me a KeyError when creating a `SigningRequest`

![image](https://github.com/user-attachments/assets/150f5c88-f331-4cf2-93ab-bc437668a4f4)

I don't see an alternative to dynamically generating the fieldset.

Reply:
you're right, I thought the admin was a bit smarter than that.

Since I found this quite ugly to have to make the fieldsets dynamic, instead I've opted to toggle the disabled state from the checkbox, so the field is now always included, but can only be edited for existing instances.